TP ESPECIAL FUNDAMENTOS DE LA CIENCIA DE DATOS 2024 

DATASET: CALIDAD DE AGUA DEL RÍO DE LA PLATA - 2022

INTEGRANTES:
Abril Valentina Valentina Juarez, Matias Müller Gonzales y Julián Elias Rivero

In [2]:
import pandas as pd

In [3]:
pd.__version__

'2.2.3'

In [4]:
raw_dataset = pd.read_csv("Calidad_de_agua_2022.csv", sep =";")

Desplegamos el dataset para realizar un vistazo generico

In [5]:
raw_dataset

orden                                 sitios codigo       fecha   año  \
0        1           Canal Villanueva y Río Luján  TI001   23/2/2022  2022   
1        2           Río Lujan y Arroyo Caraguatá  TI006   23/2/2022  2022   
2        3            Canal Aliviador y Río Lujan  TI002   23/2/2022  2022   
3        4  Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022  2022   
4        5            Río Reconquista y Río Lujan  TI004   23/2/2022  2022   
..     ...                                    ...    ...         ...   ...   
163     38                  Balneario Palo Blanco  BS092  31/10/2022  2022   
164     39          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  2022   
165     40                     Playa La Bagliardi  BS091  31/10/2022  2022   
166     41                    Balneario Municipal  BS094  31/10/2022  2022   
167     42                      Playa La Balandra  BS093  31/10/2022  2022   

       campaña tem_agua tem_aire           od           ph  ...     dbo_mg_l  \
0       Verano     24.5     23.3          5.3         6.56  ...          6.2   
1       Verano     25.4     23.3         2.25         6.56  ...          5.8   
2       Verano     24.6     23.3         2.94         6.59  ...          1.9   
3       Verano     25.2     23.3         2.22         7.45  ...          5.8   
4       Verano     24.1       20         1.02         6.39  ...          2.6   
..         ...      ...      ...          ...          ...  ...          ...   
163  Primavera       10       12  no se midió  no se midió  ...  no se midió   
164  Primavera       10       12  no se midió  no se midió  ...  no se midió   
165  Primavera       10       12  no se midió  no se midió  ...  no se midió   
166  Primavera       10       12  no se midió  no se midió  ...  no se midió   
167  Primavera       10       12  no se midió  no se midió  ...  no se midió   

    dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0         29            90                <0.10        <0.005        <0.001   
1         29            34                <0.10        <0.005        <0.001   
2         29            17                <0.10        <0.005        <0.001   
3         29            23                <0.10        <0.005        <0.001   
4         59            18                <0.10        <0.005        <0.001   
..       ...           ...                  ...           ...           ...   
163      <30            23                <0.10        <0.005        <0.001   
164       31          18.2                <0.10        <0.005        <0.001   
165      <30            40                <0.10        <0.005        <0.001   
166       39            90                <0.10             5        <0.001   
167      <30            70                <0.10             5        <0.001   

    clorofila_a_ug_l microcistina_ug_l ica             calidad_de_agua  
0                <10             <0.20  55             Muy deteriorada  
1                <10             <0.20  42  Extremadamente deteriorada  
2                <10               0.2  45             Muy deteriorada  
3                <10             <0.20  46             Muy deteriorada  
4                <10             <0.20  44  Extremadamente deteriorada  
..               ...               ...  ..                         ...  
163              2.1             <0.15  43  Extremadamente deteriorada  
164             20.2             <0.15  37  Extremadamente deteriorada  
165              0.2             <0.15  49             Muy deteriorada  
166             10.5             <0.15  39  Extremadamente deteriorada  
167             48.0             <0.15  34  Extremadamente deteriorada  

[168 rows x 31 columns]

Ahora le pedimos un poco de informacion para analizar contexto de los datos

In [6]:
raw_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168 entries, 0 to 167
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   orden                    168 non-null    int64 
 1   sitios                   168 non-null    object
 2   codigo                   168 non-null    object
 3   fecha                    168 non-null    object
 4   año                      168 non-null    object
 5   campaña                  168 non-null    object
 6   tem_agua                 168 non-null    object
 7   tem_aire                 166 non-null    object
 8   od                       168 non-null    object
 9   ph                       168 non-null    object
 10  olores                   168 non-null    object
 11  color                    168 non-null    object
 12  espumas                  168 non-null    object
 13  mat_susp                 168 non-null    object
 14  colif_fecales_ufc_100ml  167 non-null    o

En total tenemos 31 columnas (atributos) y 168 filas (observaciones). Se puede ver que hay ciertas columnas con NULL y que el tipo de datos practicamente es OBJECT exceptuando la primer columna "orden" que es de tipo INT64.

In [7]:
raw_dataset.head()

orden                                 sitios codigo      fecha   año  \
0      1           Canal Villanueva y Río Luján  TI001  23/2/2022  2022   
1      2           Río Lujan y Arroyo Caraguatá  TI006  23/2/2022  2022   
2      3            Canal Aliviador y Río Lujan  TI002  23/2/2022  2022   
3      4  Río Carapachay y Arroyo Gallo Fiambre  TI003  23/2/2022  2022   
4      5            Río Reconquista y Río Lujan  TI004  23/2/2022  2022   

  campaña tem_agua tem_aire    od    ph  ... dbo_mg_l dqo_mg_l turbiedad_ntu  \
0  Verano     24.5     23.3   5.3  6.56  ...      6.2       29            90   
1  Verano     25.4     23.3  2.25  6.56  ...      5.8       29            34   
2  Verano     24.6     23.3  2.94  6.59  ...      1.9       29            17   
3  Verano     25.2     23.3  2.22  7.45  ...      5.8       29            23   
4  Verano     24.1       20  1.02  6.39  ...      2.6       59            18   

  hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l clorofila_a_ug_l  \
0                <0.10        <0.005        <0.001              <10   
1                <0.10        <0.005        <0.001              <10   
2                <0.10        <0.005        <0.001              <10   
3                <0.10        <0.005        <0.001              <10   
4                <0.10        <0.005        <0.001              <10   

  microcistina_ug_l ica             calidad_de_agua  
0             <0.20  55             Muy deteriorada  
1             <0.20  42  Extremadamente deteriorada  
2               0.2  45             Muy deteriorada  
3             <0.20  46             Muy deteriorada  
4             <0.20  44  Extremadamente deteriorada  

[5 rows x 31 columns]

A simple vista veo que tendremos que tratar lo siguiente:
1) Hay valores que enves de tener un tipo de dato cuantitativo en el atributo, tienen un string categorico que dice "no se midio", por ejemplo en OD(oxigeno disuelto), esto deberia de ser reemplazado (ver despues con que).
2) Hay datos cuantitativos que adelante del valor numerico tienen un string "<", como en el caso de las mediciones de los contaminantes en el agua.
Podriamos de alguna manera, mediante una funcion operar y captar ese string para borrar todas las ocurrencias. 
3) Hay datos cualitativos nominales que indican "presencia" o "ausencia" de algun compuesto en el agua, pueden ser reemplazados por {0, 1} de acuerdo a su impacto en la muestra.
4) Hay datos cualitativos ordinales como "muy deteriorada" o "extremadamente deteriorada" que hacen referencia al agua de esa zona, podrian ser reemplazados mediante transformacion "ordinal encoding" ya que el orden es significativo. Algo del estilo... Rating: {low, medium, high} → {1,2,3}